src/routes/find-documents.ts

```diff
@@ -0,0 +1,24 @@
+import dependencies from '../dependencies';
+import { RouteOptions } from 'fastify';
+import { FindDocuments } from '../use-cases';
+
+interface EndpointDependencies {
+  findDocuments: FindDocuments;
+}
+
+const createEndpoint = ({
+  findDocuments,
+}: EndpointDependencies): RouteOptions => ({
+  method: 'POST',
+  url: '/documents',
+  handler: async (req, reply) => {
+    const result = await findDocuments.execute({ metadata: req.body });
+    reply.status(200).send(result);
```

Comment:
Should this always be a 200?

Reply:
I'm not sure (?) It returns a different code if there is an error tho. So like 500 etc, I guess I can write a test for that

Reply:
So if there is an error in ```findDocuments.execute()```, ```reply.status(200).send(result)``` doesn't get called?

Reply:
`findDocuments.execute` will throw in those cases (I think I saw a test to verify this?). If the handler throws an error, then fastify will catch this and by default return a 500 instead.

You can override the default error handling behaviour (https://www.fastify.io/docs/v2.0.x/Server/#seterrorhandler) if we wanted to add some custom logging etc. though.

Reply:
great, so the 200 is fine as is